src/fighter.cpp

```diff
@@ -21,72 +21,53 @@
 
 #include "fighter.h"
 
+#include "utilities.h"
+
 KFighter Fighter;
 
 KFighter::KFighter()
-    : name { "" }
-    , xp { 0 }
-    , gp { 0 }
-    , lvl { 0 }
-    , cx { 0 }
-    , cy { 0 }
-    , cw { 0 }
-    , cl { 0 }
-    , hp { 0 }
-    , mhp { 0 }
-    , mp { 0 }
-    , mmp { 0 }
-    , dip { 0 }
-    , defeat_item_common { 0 }
-    , defeat_item_rare { 0 }
-    , steal_item_common { 0 }
-    , steal_item_rare { 0 }
-    , facing { 0 }
-    , aframe { 0 }
-    , crit { 0 }
-    , defend { 0 }
-    , csmem { 0 }
-    , ctmem { 0 }
-    , current_weapon_type { 0 }
-    , welem { 0 }
-    , unl { 0 }
-    , aux { 0 }
-    , bonus { 0 }
-    , bstat { 0 }
-    , mrp { 0 }
-    , imb_s { 0 }
-    , imb_a { 0 }
-    , opal_power { 0 }
+    : name {}
+    , xp {}
+    , gp {}
+    , lvl {}
+    , cx {}
+    , cy {}
+    , cw {}
+    , cl {}
+    , hp {}
+    , mhp {}
+    , mp {}
+    , mmp {}
+    , dip {}
+    , defeat_item_common {}
+    , defeat_item_rare {}
+    , steal_item_common {}
+    , steal_item_rare {}
+    , stats {}
+    , res {}
+    , facing {}
+    , aframe {}
+    , crit {}
+    , defend {}
+    , ai {}
+    , aip {}
+    , atrack {}
```

Comment:
Turns out, default-initializing an array DOES zero the whole thing out.

Reply:
Hm i did not realise that!